gpkit/constraints/geometric_program.py

```diff
@@ -57,8 +57,14 @@ def __init__(self, cost, constraints, substitutions=None, verbosity=1):
         if not isinstance(constraints, ConstraintSet):
             constraints = ConstraintSet(constraints)
         list.__init__(self, [constraints])  # pylint:disable=non-parent-init-called
-        self.substitutions = substitutions if substitutions else {}
+        self.substitutions = substitutions or {}
         for key, sub in self.substitutions.items():
```

Comment:
GP is where the substitutions are checked to be numbers already, so I think it makes sense for them to impose the unit conversion.